ProboCI/probo-proxy
# Features/lol/improve error responses

Related to:
https://github.com/ProboCI/probo-coordinator/pull/84
and
https://github.com/ProboCI/probo/pull/68
overlaps
https://github.com/ProboCI/probo-proxy/pull/4

\We want to have the proxy return better errors for humans in some cases.
Particularly, in the case of Probo SASS, we want the probo frontend to
be redirected to a nice error page. If however there is no front end and
the user is visiting the proxy URL via a browser, they should also see a
nice(r) error page. To handle this, configuration settings were added
that allow the proxy to be configured with a redirection URL for error
handling and a string of custom 404 error HTML. We inspect the request's
headers to determine if the response should be HTML (redirection or
HTML for a browser) or JSON (cli) or a plaintext response (no accept
headers sent).

### To test
 - `npm test` and ensure all tests pass.
 - Follow tests for https://github.com/ProboCI/probo/pull/68 to set up builds that are reaped or in progress.
 - Visit the proxy URLs for those builds via your browser
   - If you have a redirectionURL set you should be redirected
   - If you have no redirectionURL you should get an HTML response
 - Visit the proxy via curl with no special headers set, you should get a plaintext response
 - Visit the proxy via curl JSON accept headers set, you should get a JSON response
